components/script/dom/webglrenderingcontext.rs

```diff
@@ -112,6 +112,11 @@ fn has_invalid_blend_constants(arg1: u32, arg2: u32) -> bool {
         (_, _) => false
     }
 }
+
+fn multiply_u8_pixel(a: u8, b: u8) -> u8 {
```

Comment:
nit: We have code for premultiplication in `components/canvas_traits/lib.rs`. They contain code similar to this, you may consider putting it there (or even all the premultiplication subroutines? Though that's not great because they aren't used anywhere else).

Anyway, your choice :)